src/components/Button/story.tsx

```diff
@@ -179,43 +181,75 @@ storiesOf('Planets/Button', module)
                   >
                     <Button level={level}>default</Button>
                   </span>
-                  {states.map((state) => (
-                    <span key={state} style={{ marginRight: '20px' }}>
-                      <Button level={level} {...{ [state]: true }}>
-                        {state}
-                      </Button>
-                    </span>
-                  ))}
+                  {[...states.map((state) => [state]), ['hover', 'focus']].map(
+                    (stateCombo) => {
+                      const key = stateCombo.join('-');
+
+                      const trueStates = stateCombo.reduce(
+                        (result, state) => ({
+                          ...result,
+                          [state]: true,
+                        }),
+                        {}
+                      );
+
+                      return (
+                        <span key={key} style={{ marginRight: '20px' }}>
+                          <Button level={level} {...trueStates}>
+                            {key}
+                          </Button>
+                        </span>
+                      );
+                    }
+                  )}
+                </div>
+              </div>
+            ))}
+          </Wrap>
+          <Wrap style={{ backgroundColor: '#00607f' }}>
+            <h3 style={{ color: 'white' }}>
+              Color variants with the `outline` prop
+            </h3>
+            {levels.map((level) => (
+              <div key={level}>
+                <h4 style={{ color: 'white' }}>{level}</h4>
+                <div style={{ marginBottom: '20px' }}>
+                  <span
+                    style={{
+                      display: 'inline-block',
+                      marginRight: '20px',
+                      marginBottom: '10px',
+                    }}
+                  >
+                    <Button level={level} outline>
+                      default
+                    </Button>
+                  </span>
+                  {[...states.map((state) => [state]), ['hover', 'focus']].map(
+                    (stateCombo) => {
+                      const key = stateCombo.join('-');
+
+                      const trueStates = stateCombo.reduce(
+                        (result, state) => ({
+                          ...result,
+                          [state]: true,
+                        }),
+                        {}
+                      );
```

Comment:
another way to do this might be something like this:
```javascript
const trueStates = Object.fromEntries(
    stateCombo.map((state) => [state, true])
  );
```
not sure why, but I always need to stare at `.reduce`'s for a while to grok them. but your way feels very React-y

Reply:
You're right, it's really hard to read. I'll take a crack at refactoring. I didn't even know about `fromEntries`, I have a feeling we're going to be friends.